src/runtime_src/driver/xclng/drm/xocl/subdev/xmc.c

```diff
@@ -168,11 +184,44 @@ struct xocl_xmc {
 	u64			cache_expire_secs;
 	struct xcl_sensor	cache;
 	ktime_t			cache_expires;
+  bool  runtime_cs_enabled; //Runtime clock scaling enabled status
 };
 
 
 static int load_xmc(struct xocl_xmc *xmc);
 static int stop_xmc(struct platform_device *pdev);
+static void xmc_clk_scale_config(struct platform_device *pdev);
+
+void xmc_set_threshold_values(struct platform_device *pdev, void *buf)
+{
+  struct xocl_xmc *xmc = platform_get_drvdata(pdev);
+  struct xclmgmt_ioc_threshold *obj = (struct xclmgmt_ioc_threshold *)buf;
+  u32 ctrl;
+
+  if (!xmc) {
+    xocl_err(&pdev->dev, "failed since xmc handle is null\n");
+    return;
+  }
+
+  /* Check if clock scaling feature enabled */
+  if (!xmc->runtime_cs_enabled) {
+    xocl_err(&pdev->dev, "failed since runtime clock scaling feature is not supported\n");
+    return;
+  }
+
+  xocl_info(&pdev->dev, "configure threshold values\n");
+  //Set Target Power
+  ctrl = READ_CLK(xmc, XMC_CLOCK_SCALING_POWER_REG);
+  ctrl &= ~XMC_CLOCK_SCALING_POWER_REG_MASK;
+  ctrl |= (obj->power & XMC_CLOCK_SCALING_POWER_REG_MASK);
+  WRITE_CLK(xmc, ctrl, XMC_CLOCK_SCALING_POWER_REG);
+
+  //Set Target Temperature
+  ctrl = READ_CLK(xmc, XMC_CLOCK_SCALING_TEMP_REG);
+  ctrl &= ~XMC_CLOCK_SCALING_TEMP_REG_MASK;
+  ctrl |= (obj->temperature & XMC_CLOCK_SCALING_TEMP_REG_MASK);
+  WRITE_CLK(xmc, ctrl, XMC_CLOCK_SCALING_TEMP_REG);
+}
```

Comment:
Yes, it's an open action item. I forgot to add TODO statement as a comment at this place in the code. I updated TODO comment here now. The power/temperature spec limits has to be read from featureROM field (which is being tracked in CR-1029672. Once the featureROM changes are ready, I will add the check to prevent setting up power/temp values more than spec limits.